autoload/ale/rename.vim

```diff
@@ -0,0 +1,193 @@
+let s:rename_map = {}
+
+" Used to get the rename map in tests.
+function! ale#rename#GetMap() abort
+    return deepcopy(s:rename_map)
+endfunction
+
+" Used to set the rename map in tests.
+function! ale#rename#SetMap(map) abort
+    let s:rename_map = a:map
+endfunction
+
+function! ale#rename#ClearLSPData() abort
+    let s:rename_map = {}
+endfunction
+
+function! ale#rename#HandleTSServerResponse(conn_id, response) abort
+    if get(a:response, 'command', '') is# 'rename'
+    \&& has_key(s:rename_map, a:response.request_seq)
+        let l:new_name = s:rename_map[a:response.request_seq].new_name
+        call remove(s:rename_map, a:response.request_seq)
+
+        if get(a:response, 'success', v:false) is v:true
+            let l:changes = []
+
+            for l:response_item in a:response.body.locs
+                let l:filename = l:response_item.file
+                let l:text_changes = []
+
+                for l:loc in l:response_item.locs
+                    call add(l:text_changes, {
+                    \ 'start': {
+                    \   'line': l:loc.start.line,
+                    \   'offset': l:loc.start.offset,
+                    \ },
+                    \ 'end': {
+                    \   'line': l:loc.end.line,
+                    \   'offset': l:loc.end.offset,
+                    \ },
+                    \ 'newText': l:new_name,
+                    \})
+                endfor
+
+                call add(l:changes, {
+                \   'fileName': l:filename,
+                \   'textChanges': l:text_changes,
+                \})
+            endfor
+
+            if empty(l:changes)
+                call ale#util#Execute('echom ''Could not rename.''')
+            else
+                call ale#code_action#HandleCodeAction({
+                \ 'description': 'rename',
+                \ 'changes': l:changes,
+                \})
+            endif
+        endif
+    endif
+endfunction
+
+function! ale#rename#HandleLSPResponse(conn_id, response) abort
+    if has_key(a:response, 'id')
+    \&& has_key(s:rename_map, a:response.id)
+        call remove(s:rename_map, a:response.id)
+
+        if !has_key(a:response, 'result')
+            call ale#util#Execute('echom ''Could not rename.''')
+
+            return
+        endif
+
+        let l:workspace_edit = a:response.result
+
+        if !has_key(l:workspace_edit, 'changes')
+            call ale#util#Execute('echom ''Could not rename.''')
```

Comment:
Write a script function for printing this message, so you don't have to write the same message several times throughout the file.